File: algorithm/python/ds/Array/35.Search Insert Position.py
# https://leetcode.com/problems/spiral-matrix/description/


def searchInsert(nums, target):
    i = 0
    j = len(nums)

    if target <= nums[0]:
        return 0
    elif target > nums[-1]:
        return j
    
    while i < j:
        mid = (i + j) // 2
        if nums[mid] >= target and nums[mid-1] < target:
            break
        elif nums[mid] < target:
            j = mid
        else: 
            i = mid + 1
    return mid




# positive 
print(searchInsert([1,3,5,6], 5))

File: algorithm/javascript/ds/Array/35.Search Insert Position.js
// Two Sum
// Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
//https://leetcode.com/problems/two-sum/description/

const searchInsert = (nums, target) => {  
    let left = 0, right = nums.length;
    while(left <= right){
        let mid = left + right >> 1;
        if(nums[mid] == target) return mid
        if(nums[mid] < target)
            left = mid + 1;
        else
            right = mid - 1;
        
    }
    return left;
}

console.log(searchInsert([1,3,5,6], 5)); // returns [3, 4, 7, 10, 12]